src/app/onprem/license-nag-message-management/nag-message.service.js

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2015-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ */
+'use strict';
+
+
+/**
+* This class is handling the service for the nag message
+* @author Oleksii Orel
+*/
+export class NagMessageService {
+
+  /**
+   * Default constructor that is using resource injection
+   * @ngInject for Dependency injection
+   */
+  constructor($document, cheBodyInjectorSvc , codenvyLicense) {
+    'ngInject';
+
+    this.$document = $document;
+    this.cheBodyInjectorSvc = cheBodyInjectorSvc;
+    this.codenvyLicense = codenvyLicense;
+
+  }
+
+  /**
+   * Show nag message
+   */
+  showNagMessage() {
+    this.$document.find('body').addClass('nag-message-indent');
```

Comment:
but high level service couldn't touch low level "body" as you're introducing a service for handling low level body element

cheBodyInjectorSvc.addClass('my-custom-class')
cheBodyInjectorSvc.insertElement('<h1>hello</h1>');

about the "exceptional situation", If you already need a way to tweak the body style, I think every nag message will have this requirement as well, no ?
.